ext/native/thin3d/thin3d_vulkan.cpp

```diff
@@ -1277,18 +1350,124 @@ uint32_t VKContext::GetDataFormatSupport(DataFormat fmt) const {
+	if (color) {
+		ici.usage |= VK_IMAGE_USAGE_COLOR_ATTACHMENT_BIT;
+	} else {
+		ici.usage |= VK_IMAGE_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT;
+	}
+	vkCreateImage(vulkan->GetDevice(), &ici, nullptr, &img.image);
+
+	VkMemoryRequirements memreq;
+	vkGetImageMemoryRequirements(vulkan->GetDevice(), img.image, &memreq);
+	
+	VkMemoryAllocateInfo alloc{ VK_STRUCTURE_TYPE_MEMORY_ALLOCATE_INFO };
+	alloc.allocationSize = memreq.size;
+	vulkan->MemoryTypeFromProperties(memreq.memoryTypeBits, VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT, &alloc.memoryTypeIndex);
+	VkResult res = vkAllocateMemory(vulkan->GetDevice(), &alloc, nullptr, &img.memory);
+	assert(res);
+	res = vkBindImageMemory(vulkan->GetDevice(), img.image, img.memory, 0);
+	assert(res);
+	img.layout = initialLayout;
+
+	VkImageViewCreateInfo ivci{ VK_STRUCTURE_TYPE_IMAGE_VIEW_CREATE_INFO };
+	ivci.components = { VK_COMPONENT_SWIZZLE_R, VK_COMPONENT_SWIZZLE_G, VK_COMPONENT_SWIZZLE_B, VK_COMPONENT_SWIZZLE_A };
+	ivci.format = ici.format;
+	ivci.image = img.image;
+	ivci.viewType = VK_IMAGE_VIEW_TYPE_2D;
+	ivci.subresourceRange.aspectMask = color ? VK_IMAGE_ASPECT_COLOR_BIT : (VK_IMAGE_ASPECT_DEPTH_BIT | VK_IMAGE_ASPECT_STENCIL_BIT);
+	ivci.subresourceRange.layerCount = 1;
+	ivci.subresourceRange.levelCount = 1;
+	res = vkCreateImageView(vulkan->GetDevice(), &ivci, nullptr, &img.view);
+	assert(res);
+
+	VkImageMemoryBarrier barrier{};
+	barrier.sType = VK_STRUCTURE_TYPE_IMAGE_MEMORY_BARRIER;
+	barrier.oldLayout = VK_IMAGE_LAYOUT_UNDEFINED;
+	barrier.subresourceRange.layerCount = 1;
+	barrier.subresourceRange.levelCount = 1;
+	barrier.image = img.image;
+	barrier.srcAccessMask = 0;
+	switch (initialLayout) {
+	case VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL:
+		barrier.dstAccessMask = VK_ACCESS_COLOR_ATTACHMENT_WRITE_BIT;
+		break;
+	case VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL:
+		barrier.dstAccessMask = VK_ACCESS_TRANSFER_WRITE_BIT;
+		break;
+	case VK_IMAGE_LAYOUT_DEPTH_STENCIL_ATTACHMENT_OPTIMAL:
+		barrier.dstAccessMask = VK_ACCESS_DEPTH_STENCIL_ATTACHMENT_WRITE_BIT;
+		break;
+	}
+	barrier.newLayout = initialLayout;
+	barrier.subresourceRange.aspectMask = ivci.subresourceRange.aspectMask;
+	vkCmdPipelineBarrier(vulkan->GetInitCommandBuffer(), VK_PIPELINE_STAGE_ALL_COMMANDS_BIT, VK_PIPELINE_STAGE_TOP_OF_PIPE_BIT, 0, 0, 0, 0, 0, 1, &barrier);
+}
+
 // A VKFramebuffer is a VkFramebuffer plus all the textures it owns.
 class VKFramebuffer : public Framebuffer {
 public:
+	VKFramebuffer(VulkanContext *vk) : vulkan_(vk) {}
+	~VKFramebuffer() {
+		vulkan_->Delete().QueueDeleteImage(color.image);
+		vulkan_->Delete().QueueDeleteImage(depth.image);
+		vulkan_->Delete().QueueDeleteImageView(color.view);
+		vulkan_->Delete().QueueDeleteImageView(depth.view);
+		vulkan_->Delete().QueueDeleteDeviceMemory(color.memory);
+		vulkan_->Delete().QueueDeleteDeviceMemory(depth.memory);
+		vulkan_->Delete().QueueDeleteFramebuffer(framebuf);
+	}
+	VkFramebuffer framebuf;
+	VKImage color;
+	VKImage depth;
```

Comment:
Worth initing to NULL or null checking above (not sure if necessary to)?

I remember we had some FBO creates that didn't need stencil/depth, not sure if we still support that or if it matters (I'm thinking of depal FBOs.)

-[Unknown]